### DISCUSSION TITLE
Use fly to find resource versions used in a build?

### DISCUSSION THREAD

**Discussion by dolph** [concourse/concourse#7550] [Help & Support]
Is there a way to view the resource metadata retrieved as `get` tasks for a given build? The web UI obviously has access to that data, but I can't really find any task-level data in `fly`, which makes me think I'm overlooking an entire command or option.

I'm on Concourse / fly v7.4.0.

**Comment by taylorsilva** [comment]
Looks like `fly resource-versions` with the `--json` flag has the metadata for each resource version in it.
```json
[ ...
{
    "id": 1202084830,
    "metadata": [
      {
        "name": "commit",
        "value": "d51308dc3747704e673151c7314cc486d1980467"
      },
      {
        "name": "author",
        "value": "Taylor Silva"
      },
      {
        "name": "author_date",
        "value": "2021-08-30 13:35:11 -0400"
      },
      {
        "name": "branch",
        "value": "master"
      },
      {
        "name": "message",
        "value": "Merge pull request #7461 from concourse/issue/7430\n\nworker: delete btrfs volume if it exists when using the overlay driver"
      },
      {
        "name": "url",
        "value": "https://github.com/concourse/concourse/commit/d51308dc3747704e673151c7314cc486d1980467"
      }
    ],
    "version": {
      "ref": "d51308dc3747704e673151c7314cc486d1980467"
    },
    "enabled": true
  }
]
```

**Reply by dolph** [reply]
Awesome! That's definitely half the answer (I wasn't aware of what was returned via `--json`). But how do I know which builds used that resource version? I think I'm expecting a `fly get-build` (or something) to get a build's "configuration" (i.e. inputs) akin to:
```
  get-pipeline               Get a pipeline's current configuration (aliases: gp)
  get-team                   Show team configuration (aliases: gt)
```

**Reply by taylorsilva** [reply]
I don't think there's a fly command to get that info. You'll need to poke at the API to get that info (which we don't document because we want to discourage people from using it)

Looks like `https://your-ci.org/api/v1/teams/<team-name>/pipelines/<pipeline>/resources/<resource>/versions/<version id>/input_to` is the API call you want.

**Reply by taylorsilva** [reply]
There's also a resource causality view in the web UI. See the 7.4.0 release notes: https://github.com/concourse/concourse/releases/tag/v7.4.0
You do need to enable it via a feature flag, `CONCOURSE_ENABLE_RESOURCE_CAUSALITY`
